scipy/optimize/optimize.py

```diff
@@ -2048,16 +2051,16 @@ def golden(func, args=(), brack=None, tol=_epsilon, full_output=0):
     interval.
 
     """
-    options = {'xtol': tol}
+    options = {'xtol': tol, 'maxiter': maxiter}
     res = _minimize_scalar_golden(func, brack, args, **options)
     if full_output:
-        return res['x'], res['fun'], res['nfev']
+        return res['x'], res['fun'], res['nit'], res['nfev']
```

Comment:
Yurii Shevchuk a écrit :
> In scipy/optimize/optimize.py
> <https://github.com/scipy/scipy/pull/6489#discussion_r74816341>:
>
>>      res = _minimize_scalar_golden(func, brack, args, **options)
>>      if full_output:
>> -        return res['x'], res['fun'], res['nfev']
>> +        return res['x'], res['fun'], res['nit'], res['nfev']
>
> Not sure that I understand this one. Could you clarify it?
>

I mean this hunk breaks bw compat and should be discarded. The other
part (modification of OptimizeResult) is fine.